sourcegraph/sourcegraph
# Code Insights: Migrate dashboard chart cards to new chart and card components API

Based on https://github.com/sourcegraph/sourcegraph/pull/33676
Closes https://github.com/sourcegraph/sourcegraph/issues/32944

## Background

In this PR we migrate our charts on the dashboard page. This implies that we don't use anymore shared model around charts from the 'sourcegraph' package that we used to have when we were dealing with extension API in code insights.

Since this PR is turned out a bit bigger than I expected here is some important parts (comments) of this PR
- [Migration Built In (runtime) insight card ](https://github.com/sourcegraph/sourcegraph/pull/33799/commits/11c2d1a6e7f9c2b4e6a26b7205ff5e2edf0596c0)
- [Improvements around fetching types in generic insight fetcher hook](https://github.com/sourcegraph/sourcegraph/pull/33799/commits/3660d49699a5ec0708f43fc4c6843a13170f1184)`
- [Adjust backend insight chart data deserialisation](https://github.com/sourcegraph/sourcegraph/pull/33799/commits/d16d732e0494b9e956f188992011b4aee2cc0b00)
- [Migrate Backend insight card ](https://github.com/sourcegraph/sourcegraph/pull/33799/commits/a576f0d5301eb3f001698136cb1452fdb962e6ab)

## Test plan
- Make sure that all dashboard charts (lang stats, search based, capture group) work properly in different browsers (Safari, Chrome, Firefox)
- Make sure that datum links work properly and have no regression 
- Make sure that dashboard looks correct on different viewport sizes


## App preview:

- [Link](https://sg-web-vk-migrate-dashboard-chart-cards.onrender.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
